### PR TITLE
Add support to create, update and delete pages

### PR DESCRIPTION
**Pull request by xavierbarbosa**
### Added
- Support to create, update and delete pages

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/pages/index.js [hunk 1]
```diff
@@ -296,13 +296,27 @@ const Pages = () => {
                 Cell: settings => {
                     return (
                         <>
-                            <a href={`/pages/${settings.original.slug}`}>
+                            <Link to={`/pages/${settings.original.slug}`}>
```
Comment (xavierbarbosa) [inline hat/assets/js/apps/Iaso/domains/pages/index.js:299]: @madewulf I added a view inside the react app which renders the dashboard inside an iframe, so we only need to add the link when creating the page. This way we keep the navigation consistent, see:

image

If you prefer the old approach we can change it back, but I do think this is nicer.

**Comment by beygorghor** [comment]
Added:
- use feature flag for page activation
- change icon menu
- adapt modale size, padding and scroll
- page type : iframe / html /raw
- use a wysiwyg tool to allow editing html
- change link to django template 
-  adding needs_auth, users to CRUD
- change view.py to use different types
- remove view page
-  split codeinto components
Screenshot 2021-08-05 at 23 39 48
Screenshot 2021-08-05 at 23 38 09
Screenshot 2021-08-05 at 23 40 03
Screenshot 2021-08-05 at 23 39 55

File: hat/templates/iaso/pages/iframe.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE HTML>
+<html>
+
```
Comment (olethanh) [inline hat/templates/iaso/pages/iframe.html:3]: maybe a title?

File: hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        }));
+    const classes = useStyles();
+    const intl = useSafeIntl();
+
+    const handleSubmit = (values, helpers) => {
+        const tempValues = { ...values };
+        const users = values.users || [];
+        users.push(currentUser.user_id);
+        tempValues.users = users;
+        savePage(tempValues, {
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js:63]: Line 61: there's a potential mutation of `values`. 
Suggested change: `tempValues.users = [...(values.users??[]), currentUser.user_id]`, in place of lines 61-63

File: hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                onClose();
+                helpers.resetForm();
+            },
+        });
+    };
+
+    const defaultValues = {
+        type: selectedPage && selectedPage.type ? selectedPage.type : RAW,
+        needs_authentication: !(
+            selectedPage && selectedPage.needs_authentication === false
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js:74]: cf line 72

File: hat/assets/js/apps/Iaso/domains/pages/index.js [hunk 5]
```diff
@@ -251,11 +87,11 @@ const Pages = () => {
     const { data: pages = [], status } = query;
 
     const selectedPage = pages?.results?.find(
-        result => result.id === selectedPageId,
+        result => result.slug === selectedPageSlug,
     );
 
     const handleDeleteConfirmDialogConfirm = () => {
-        removePage(selectedPage.id, {
+        removePage(selectedPage.slug, {
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/pages/index.js:94]: this will throw an error if selectedPage is undefined -> `selectedPage?.slug`

**Review by quang-le — CHANGES_REQUESTED** [review]
-  remove potential mutation in `CreateEditDialog` line 60.
-  remove potential null error in `index.js`  line 94
- other comments are more stylistic suggestions

Nice work, great feature!

File: hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js (hunk 3 above)
Comment (olethanh) [reply hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js:63]: not really important but shouldn't we check the currentUser is not already included before adding it?

Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/pages/components/CreateEditDialog.js:63]: Without a currentUser, the page is not loaded

File: hat/assets/js/apps/Iaso/components/dialogs/ConfirmDialogComponent.js [hunk 6]
```diff
@@ -46,7 +46,14 @@ const ConfirmDialog = ({
...
             </Button>
-            <Dialog open={open} onClick={() => handleClose(false)}>
+            <Dialog
+                open={open}
+                onClose={(event, reason) => {
+                    if (reason === 'backdropClick') {
+                        handleClose(false);
+                    }
+                }}
+            >
```
[the remark this answers is not in the document]
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/components/dialogs/ConfirmDialogComponent.js:56]: otherwhise the dialog will close on any click... lika a focus on a input

**Pull request merged**